marshmallow/fields.py

```diff
@@ -994,6 +996,31 @@ def _deserialize(self, value, attr, data):
             self.fail('invalid')
 
 
+class Dict(Field):
+    """A dict field. Supports dicts and dict-like objects.
+
+    .. versionadded:: 2.1.0
+    """
+
+    default_error_messages = {
+        'invalid': 'Not a valid dict.'
+    }
+
+    def _validated(self, value):
+        if isinstance(value, collections.Mapping):
+            return value
+        else:
+            self.fail('invalid')
+
+    def _serialize(self, value, attr, obj):
+        if value is None:
+            return None
+        return self._validated(value)
```

Comment:
I was modeling it after `fields.Number`, which validates the type during serialization. It will try to turn the thing into a float.

In fact I may want to `return dict(value)` here, to convert dict-like things into true Python dictionaries. Marshmallow is for converting complex data types to primitive Python data types. This would ease any further serialization, such as jsonifying it.